mlflow/utils/environment.py

```diff
@@ -666,6 +667,30 @@ def _deduplicate_requirements(requirements):
     return [str(req) for req in deduped_reqs.values()]
 
 
+def _fetch_requirement_name(req: str) -> str:
```

Comment:
```suggestion
def _parse_requirement_name(req: str) -> str:
```

Let's rename this. `fetch` sounds like we make an HTTP request to fetch the name.